src/LinAlg/hiopVectorRajaImpl.hpp

```diff
@@ -161,48 +162,45 @@ hiopVectorRajaPar::hiopVectorRajaPar(const hiopVectorRajaPar& v)
   mem_space_ = v.mem_space_;
 
 #ifndef HIOP_USE_GPU
-  mem_space_ = "HOST"; // If no GPU support, fall back to host!
+  assert(mem_space_ == "HOST"); 
 #endif
 
-  // std::cout << "Memory space: " << mem_space_ << "\n";
-  auto& resmgr = umpire::ResourceManager::getInstance();
-  umpire::Allocator devalloc  = resmgr.getAllocator(mem_space_);
-  data_dev_ = static_cast<double*>(devalloc.allocate(n_local_*sizeof(double)));
-  if(mem_space_ == "DEVICE")
+  data_dev_ = exec_space_.template alloc_array<double>(n_local_);
+  if(exec_space_.mem_backend().is_device())
   {
     // Create host mirror if the memory space is on device
-    umpire::Allocator hostalloc = resmgr.getAllocator("HOST");
-    data_host_ = static_cast<double*>(hostalloc.allocate(n_local_*sizeof(double)));
+    data_host_ = exec_space_host_.template alloc_array<double>(n_local_);    
   }
   else
   {
     data_host_ = data_dev_;
   }
 }
 
-hiopVectorRajaPar::~hiopVectorRajaPar()
+template<class MEMBACKEND, class RAJACUDAPOL>
+hiopVectorRaja<MEMBACKEND,RAJACUDAPOL>::~hiopVectorRaja()
 {
-  auto& resmgr = umpire::ResourceManager::getInstance();
-  umpire::Allocator devalloc  = resmgr.getAllocator(mem_space_);
-  if(data_dev_ != data_host_)
-  {
-    umpire::Allocator hostalloc = resmgr.getAllocator("HOST");
-    hostalloc.deallocate(data_host_);
+  if(data_dev_ != data_host_) {
+    exec_space_host_.dealloc_array(data_host_);
   }
-  devalloc.deallocate(data_dev_);
+  exec_space_.dealloc_array(data_dev_);
   data_dev_  = nullptr;
   data_host_ = nullptr;
   delete idx_cumsum_;
 }
 
-hiopVector* hiopVectorRajaPar::alloc_clone() const
+template<class MEM, class POL>
+hiopVector* hiopVectorRaja<MEM,POL>::alloc_clone() const
 {
-  hiopVector* v = new hiopVectorRajaPar(*this); assert(v);
+  hiopVector* v = new hiopVectorRaja<MEM,POL>(*this);
+  assert(v);
```

Comment:
Is this necessary? I would hope that `hiopVectorRaja`'s constructor would handle error checking